### PR TITLE
python3 migration: convert iterator to list every time we delete elements of the corresponding object

### PR DESCRIPTION
**Pull request by philipwinklerdelmic**

### CODE REVIEW

File: src/odemis/cli/main.py [hunk 1]
```diff
@@ -110,8 +110,8 @@ def scan(cls=None):
                 else:
                     if not devices:
                         logging.info("No device found")
-                    for name, args in devices:
-                        print("%s.%s: '%s' init=%r" % (module_name, cls_name, name, str(args)))
+                    #for name, args in devices:
+                    #    print("%s.%s: '%s' init=%r" % (module_name, cls_name, name, args))
```
Comment (pieleric) [inline src/odemis/cli/main.py]: That change is not related, and probably shouldn't be done anyway.

File: src/odemis/util/img.py [hunk 2]
```diff
@@ -668,7 +668,7 @@ def mergeMetadata(current, correction=None):
 
     # remove correction metadata (to make it clear the correction has been applied)
     for k in (model.MD_ROTATION_COR, model.MD_PIXEL_SIZE_COR, model.MD_POS_COR, model.MD_SHEAR_COR):
-        if k in current:
+        if k in list(current):
```
Comment (pieleric) [inline src/odemis/util/img.py]: No, not in a `if`.

File: src/odemis/odemisd/modelgen.py [hunk 3]
```diff
@@ -730,7 +730,7 @@ def _update_properties(self, comp_name):
                                     "Component '%s' has no property '%s' (defined as persistent)." % (comp_name, prop_name))
 
             # Load value from settings file if available
-            if prop_name in persistent_props_values:
+            if prop_name in list(persistent_props_values):
```
Comment (pieleric) [inline src/odemis/odemisd/modelgen.py]: No, not in a `if`, for the two changes in this file.

File: src/odemis/odemisd/main.py [hunk 4]
```diff
@@ -404,7 +404,7 @@ def _terminate_component(self, c):
         # was created.
         # TODO: check there is really no component still running in the
         # container?
-        if cname in self._instantiator.sub_containers:
+        if cname in list(self._instantiator.sub_containers):
```
Comment (pieleric) [inline src/odemis/odemisd/main.py]: No, not in a `if`.

**Comment by philipwinklerdelmic** [comment]
updated
